src/globus_sdk/services/transfer/transport.py

```diff
@@ -0,0 +1,25 @@
+"""
+Custom Transport class for the TransferClient that overrides
+default_check_transient_error
+"""
+from globus_sdk.transport import RequestsTransport, RetryCheckResult, RetryContext
+
+
+class TransferRequestsTransport(RequestsTransport):
+    def default_check_transient_error(self, ctx: RetryContext) -> RetryCheckResult:
+        """
+        check for transient error status codes which could be resolved by
+        retrying the request. Does not retry ExternalErrors as those are
+        unlikely to actually be transient.
+        """
+        if ctx.response is not None and (
+            ctx.response.status_code in self.TRANSIENT_ERROR_STATUS_CODES
+        ):
+            try:
+                if "ExternalError" in ctx.response.json()["code"]:
```

Comment:
I feel like there's a lot happening in one line here. JSON parsing and indexing and a comparison.

Can we break it up a little, maybe like
```
try:
   code = ctx.response.json()["code"]
except (ValueError, KeyError):
    code = ""
if code.startswith("ExternalError"): ...  # also, is changing to `startswith` a good idea or not?
```

Reply:
Sure, I'll break it up
I believe the Transfer error code may be RestExternalError in some cases, so probably best to stick with `in`

Reply:
That is somewhat unfortunate, but understandable. If we need to refine this in the future we can always `code.split(".")`, and so forth.